Examples/ObjectiveC/UserTrackingModesExample.m

```diff
@@ -3,23 +3,182 @@
 
 NSString *const MBXExampleUserTrackingModes = @"UserTrackingModesExample";
 
+// Subclass UIButton to create a custom user tracking mode button
+@interface UserLocationButton : UIButton
+@property (nonatomic) CAShapeLayer *arrow;
+@end
+
+const CGFloat UserLocationButtonSize = 80;
+
+@implementation UserLocationButton
+
+- (instancetype)init {
+    self = [super initWithFrame:CGRectMake(0, 0, UserLocationButtonSize, UserLocationButtonSize)];
```

Comment:
Let’s refactor this and `layoutSubviews` to follow [a more typical pattern](https://developer.apple.com/library/content/documentation/General/Conceptual/DevPedia-CocoaCore/Initialization.html), like:

```
- (instancetype)initWithFrame:(CGRect)frame {
    if (self = [super initWithFrame:frame]) {
        [self commonInit];
    }
    return self;
}
```